triplet/main.py

```diff
@@ -151,7 +145,13 @@ def convert_and_transform(img, transform):
         return img_a, img_p, img_n
 
     def __len__(self):
-        return self.triplets.shape[0]
+        try:
+            if self.train:
+                return self.triplets.shape[0]
+            else:
+                return self.matches.shape[0]
+        except:
+            pass
 
```

Comment:
That's the error
```
File "/home/eriba/software/pytorch/examples-edgarriba/triplet/main.py", line 148, in __len__
    if self.train:
AttributeError: 'TripletPhotoTour' object has no attribute 'train'
```

I know it's weird, but I traced the error and it looks like that when the parent class is constructed it calls this method instead of it's own